// payhere-js-sdk/lib/params/PreapprovalParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PreapprovalParams = void 0;
const utils_1 = require("../utils");
const Payhere_1 = require("../Payhere");
class PreapprovalParams {
    constructor(preapprovalParams) {
        this._preapprovalParams = preapprovalParams;
    }
    toJSON() {
        const jsonObj = {
            merchant_id: Payhere_1.Payhere.merchantId,
            return_url: this._preapprovalParams.returnUrl,
            cancel_url: this._preapprovalParams.cancelUrl,
            notify_url: this._preapprovalParams.notifyUrl,
            order_id: this._preapprovalParams,
            items: this._preapprovalParams.itemTitle,
            currency: this._preapprovalParams.currency,
            platform: this._preapprovalParams.platform,
            custom_1: this._preapprovalParams.custom1,
            custom_2: this._preapprovalParams.custom2,
            hash: this._preapprovalParams.hash,
        };
        return utils_1.removeBlankAttributes(jsonObj);
    }
}
exports.PreapprovalParams = PreapprovalParams;
