// payhere-js-sdk/lib/enums/CurrencyType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CurrencyType = void 0;
var CurrencyType;
(function (CurrencyType) {
    CurrencyType["USD"] = "USD";
    CurrencyType["CAD"] = "CAD";
    CurrencyType["EUR"] = "EUR";
    CurrencyType["AED"] = "AED";
    CurrencyType["AFN"] = "AFN";
    CurrencyType["ALL"] = "ALL";
    CurrencyType["AMD"] = "AMD";
    CurrencyType["ARS"] = "ARS";
    CurrencyType["AUD"] = "AUD";
    CurrencyType["AZN"] = "AZN";
    CurrencyType["BAM"] = "BAM";
    CurrencyType["BDT"] = "BDT";
    CurrencyType["BGN"] = "BGN";
    CurrencyType["BHD"] = "BHD";
    CurrencyType["BIF"] = "BIF";
    CurrencyType["BND"] = "BND";
    CurrencyType["BOB"] = "BOB";
    CurrencyType["BRL"] = "BRL";
    CurrencyType["BWP"] = "BWP";
    CurrencyType["BYN"] = "BYN";
    CurrencyType["BZD"] = "BZD";
    CurrencyType["CDF"] = "CDF";
    CurrencyType["CHF"] = "CHF";
    CurrencyType["CLP"] = "CLP";
    CurrencyType["CNY"] = "CNY";
    CurrencyType["COP"] = "COP";
    CurrencyType["CRC"] = "CRC";
    CurrencyType["CVE"] = "CVE";
    CurrencyType["CZK"] = "CZK";
    CurrencyType["DJF"] = "DJF";
    CurrencyType["DKK"] = "DKK";
    CurrencyType["DOP"] = "DOP";
    CurrencyType["DZD"] = "DZD";
    CurrencyType["EEK"] = "EEK";
    CurrencyType["EGP"] = "EGP";
    CurrencyType["ERN"] = "ERN";
    CurrencyType["ETB"] = "ETB";
    CurrencyType["GBP"] = "GBP";
    CurrencyType["GEL"] = "GEL";
    CurrencyType["GHS"] = "GHS";
    CurrencyType["GNF"] = "GNF";
    CurrencyType["GTQ"] = "GTQ";
    CurrencyType["HKD"] = "HKD";
    CurrencyType["HNL"] = "HNL";
    CurrencyType["HRK"] = "HRK";
    CurrencyType["HUF"] = "HUF";
    CurrencyType["IDR"] = "IDR";
    CurrencyType["ILS"] = "ILS";
    CurrencyType["INR"] = "INR";
    CurrencyType["IQD"] = "IQD";
    CurrencyType["IRR"] = "IRR";
    CurrencyType["ISK"] = "ISK";
    CurrencyType["JMD"] = "JMD";
    CurrencyType["JOD"] = "JOD";
    CurrencyType["JPY"] = "JPY";
    CurrencyType["KES"] = "KES";
    CurrencyType["KHR"] = "KHR";
    CurrencyType["KMF"] = "KMF";
    CurrencyType["KRW"] = "KRW";
    CurrencyType["KWD"] = "KWD";
    CurrencyType["KZT"] = "KZT";
    CurrencyType["LBP"] = "LBP";
    CurrencyType["LKR"] = "LKR";
    CurrencyType["LTL"] = "LTL";
    CurrencyType["LVL"] = "LVL";
    CurrencyType["LYD"] = "LYD";
    CurrencyType["MAD"] = "MAD";
    CurrencyType["MDL"] = "MDL";
    CurrencyType["MGA"] = "MGA";
    CurrencyType["MKD"] = "MKD";
    CurrencyType["MMK"] = "MMK";
    CurrencyType["MOP"] = "MOP";
    CurrencyType["MUR"] = "MUR";
    CurrencyType["MXN"] = "MXN";
    CurrencyType["MYR"] = "MYR";
    CurrencyType["MZN"] = "MZN";
    CurrencyType["NAD"] = "NAD";
    CurrencyType["NGN"] = "NGN";
    CurrencyType["NIO"] = "NIO";
    CurrencyType["NOK"] = "NOK";
    CurrencyType["NPR"] = "NPR";
    CurrencyType["NZD"] = "NZD";
    CurrencyType["OMR"] = "OMR";
    CurrencyType["PAB"] = "PAB";
    CurrencyType["PEN"] = "PEN";
    CurrencyType["PHP"] = "PHP";
    CurrencyType["PKR"] = "PKR";
    CurrencyType["PLN"] = "PLN";
    CurrencyType["PYG"] = "PYG";
    CurrencyType["QAR"] = "QAR";
    CurrencyType["RON"] = "RON";
    CurrencyType["RSD"] = "RSD";
    CurrencyType["RUB"] = "RUB";
    CurrencyType["RWF"] = "RWF";
    CurrencyType["SAR"] = "SAR";
    CurrencyType["SDG"] = "SDG";
    CurrencyType["SEK"] = "SEK";
    CurrencyType["SGD"] = "SGD";
    CurrencyType["SOS"] = "SOS";
    CurrencyType["SYP"] = "SYP";
    CurrencyType["THB"] = "THB";
    CurrencyType["TND"] = "TND";
    CurrencyType["TOP"] = "TOP";
    CurrencyType["TRY"] = "TRY";
    CurrencyType["TTD"] = "TTD";
    CurrencyType["TWD"] = "TWD";
    CurrencyType["TZS"] = "TZS";
    CurrencyType["UAH"] = "UAH";
    CurrencyType["UGX"] = "UGX";
    CurrencyType["UYU"] = "UYU";
    CurrencyType["UZS"] = "UZS";
    CurrencyType["VEF"] = "VEF";
    CurrencyType["VND"] = "VND";
    CurrencyType["XAF"] = "XAF";
    CurrencyType["XOF"] = "XOF";
    CurrencyType["YER"] = "YER";
    CurrencyType["ZAR"] = "ZAR";
    CurrencyType["ZMK"] = "ZMK";
    CurrencyType["ZWL"] = "ZWL";
})(CurrencyType = exports.CurrencyType || (exports.CurrencyType = {}));
