// payhere-js-sdk/lib/params/CheckoutParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutParams = void 0;
const utils_1 = require("../utils");
const Payhere_1 = require("../Payhere");
class CheckoutParams {
    constructor(checkoutParams) {
        this._checkoutParams = checkoutParams;
    }
    toJSON() {
        const formattedItems = this._checkoutParams.items
            ? this._checkoutParams.items.map((item, index) => ({
                [`item_name_${index}`]: item.name,
                [`item_number_${index}`]: item.modelNo,
                [`amount_${index}`]: item.amount,
                [`quantity_${index}`]: item.quantity,
            }))
            : [];
        const jsonObj = {
            merchant_id: Payhere_1.Payhere.merchantId,
            return_url: this._checkoutParams.returnUrl,
            cancel_url: this._checkoutParams.cancelUrl,
            notify_url: this._checkoutParams.notifyUrl,
            order_id: this._checkoutParams.order_id,
            items: this._checkoutParams.itemTitle,
            currency: this._checkoutParams.currency,
            amount: this._checkoutParams.amount,
            ...formattedItems,
            platform: this._checkoutParams.platform,
            custom_1: this._checkoutParams.custom1,
            custom_2: this._checkoutParams.custom2,
            hash: this._checkoutParams.hash,
        };
        return utils_1.removeBlankAttributes(jsonObj);
    }
}
exports.CheckoutParams = CheckoutParams;
