// payhere-js-sdk/lib/PayhereSubscription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayhereSubscription = void 0;
const Payhere_1 = require("./Payhere");
const utils_1 = require("./utils");
const requiredSubscriptionParams = {
    return_url: 'returnUrl',
    cancel_url: 'cancelUrl',
    notify_url: 'notifyUrl',
    first_name: 'firstName',
    last_name: 'lastName',
    email: 'email',
    phone: 'phone',
    address: 'address',
    city: 'city',
    country: 'country',
    order_id: 'orderId',
    items: 'itemTitle',
    currency: 'currency',
    recurrence: 'recurrence',
    duration: 'duration',
    amount: 'amount',
};
class PayhereSubscription extends Payhere_1.Payhere {
    constructor(customer, subscriptionParams, onError) {
        super();
        this.onSubscriptionError = (errorMsg) => console.log(errorMsg);
        try {
            this._customerInfo = customer;
            this._subscriptionParams = subscriptionParams;
            this.onSubscriptionError = onError;
        }
        catch (err) {
            throw new Error(err);
        }
    }
    async start() {
        try {
            const subscriptionReq = {
                ...this._customerInfo.toJSON(),
                ...this._subscriptionParams.toJSON(),
            };
            utils_1.submitPayhereClientRequest(subscriptionReq, requiredSubscriptionParams, '/pay/checkout', this.onSubscriptionError);
        }
        catch (err) {
            console.error(err);
            this.onSubscriptionError('Unknown error: ' + err);
        }
    }
}
exports.PayhereSubscription = PayhereSubscription;
