// payhere-js-sdk/lib/PayhereCheckout.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayhereCheckout = void 0;
const Payhere_1 = require("./Payhere");
const utils_1 = require("./utils");
const requiredCheckoutParams = {
    return_url: 'returnUrl',
    cancel_url: 'cancelUrl',
    notify_url: 'notifyUrl',
    first_name: 'firstName',
    last_name: 'lastName',
    email: 'email',
    phone: 'phone',
    address: 'address',
    city: 'city',
    country: 'country',
    order_id: 'orderId',
    items: 'itemTitle',
    currency: 'currency',
    amount: 'amount',
};
class PayhereCheckout extends Payhere_1.Payhere {
    constructor(customer, checkoutParams, onError) {
        super();
        this.onCheckoutError = (errorMsg) => console.error;
        try {
            if (!customer) {
                throw new Error("Required parameter 'customer' is missing");
            }
            if (!checkoutParams) {
                throw new Error("Required parameter 'checkoutData' is missing");
            }
            this._customerData = customer;
            this._checkoutParams = checkoutParams;
            if (onError) {
                this.onCheckoutError = onError;
            }
        }
        catch (err) {
            this.onCheckoutError(err);
        }
    }
    start() {
        try {
            const checkoutReq = {
                ...this._checkoutParams.toJSON(),
                ...this._customerData.toJSON(),
            };
            utils_1.submitPayhereClientRequest(checkoutReq, requiredCheckoutParams, '/pay/checkout', this.onCheckoutError);
        }
        catch (err) {
            this.onCheckoutError('Unknown error: ' + err);
        }
    }
}
exports.PayhereCheckout = PayhereCheckout;
